backend/src/test/java/com/github/hamideh6182/controller/BookControllerTest.java

```diff
@@ -79,6 +79,26 @@ void getAllBooks_WhenOneBookInRepo_ThenReturnListOfOneBook() throws Exception {
                         """));
     }
 
+    @Test
+    @DirtiesContext
+    void getBookByIdTest() throws Exception {
+        bookRepository.save(book1);
+        mockMvc.perform(MockMvcRequestBuilders.get("/api/books/1"))
+                .andExpect(status().isOk())
+                .andExpect(content().json("""
+                                                {
+                                                "id" : "1",
+                                                "title" : "JavaBook",
+                                                "author" : "Hamideh Aghdam",
+                                                "description" : "About Java",
+                                                "copies" : 10,
+                                                "copiesAvailable" : 10,
+                                                "category" : "Programming",
+                                                "img" : "http://imgage.com/img1.png"
+                                                }
+                        """));
+    }
+
```

Comment:
I would suggest to rename the test with the when_then scheme, like you did in the Service tests. ;)